docs/src/howto-guides/029_mqtt_local_broker_authentication.md

```diff
@@ -0,0 +1,166 @@
+# Use MQTT authentication for local broker
+
+thin-edge.io supports using certificate-based authentication when communicating
+with a local broker. There are 3 ways thin-edge.io can use when connecting to
+Mosquitto broker:
+
+1. No authentication (default)
+2. Server authentication
+3. Server + client authentication
+
+## Server authentication
+
+Enabling server authentication causes thin-edge.io MQTT clients to require a
+valid certificate from a broker when connecting. The broker certificate is valid
+when it is signed by a CA that the clients trust.
+
+To enable server authentication, perform the following:
+
+### Step 1: Configure server authenticated listener in Mosquitto broker
+
+Create a file in `/etc/mosquitto/conf.d/` with the following content and restart
+mosquitto service:
+
+```conf
+listener 8883
+certfile PATH_TO_SERVER_CERTIFICATE
+keyfile  PATH_TO_SERVER_PRIVATE_KEY
+```
+
+- `listener 8883`: defines a new listener on port 8883
+- `certfile`: points to a certificate that will be used by the broker to
+  authenticate itself to connecting clients
+- `keyfile`: points to a private key of the specified certificate, necessary for
+  encrypted communication
```

Comment:
What is document here is correct. However, I would like to stress that preparing such a configuration file is the point of the `mqtt.external.*` settings of `tedge config`.

```
tedge config list --doc | grep mqtt.external    
mqtt.external.port             Mqtt broker port, which is used by the external mqtt clients to publish or subscribe. Example: 8883
mqtt.external.bind_address     IP address / hostname, which the mqtt broker limits incoming connections on. Example: 0.0.0.0
mqtt.external.bind_interface   Name of network interface, which the mqtt broker limits incoming connections on. Example: wlan0
mqtt.external.capath           Path to a file containing the PEM encoded CA certificates that are trusted when checking incoming client certificates. Example: /etc/ssl/certsNote: If the capath is not set, then no certificates are required for the external connections.
mqtt.external.certfile         Path to the certificate file, which is used by external MQTT listenerExample: /etc/tedge/device-certs/tedge-certificate.pemNote: This setting shall be used together with `mqtt.external.keyfile` for external connections.
mqtt.external.keyfile          Path to the private key file, which is used by external MQTT listenerExample: /etc/tedge/device-certs/tedge-private-key.pemNote: This setting shall be used together with `mqtt.external.certfile` for external connections.
```

- See https://thin-edge.github.io/thin-edge.io/html/howto-guides/013_connect_external_device.html.
- @Bravo555  Change nothing for your current PR. In a follow-up task, we will have to assess if something need to be change for this `mqtt.external` setting.
- @reubenmiller I think this `mqtt.external` setting is badly named making things confusing.

Reply:
> @reubenmiller I think this mqtt.external setting is badly named making things confusing.

Yes I fully agree. It is really just a binding address, it does not matter if it is "internal" or "external"...but yes it can be done as a later stage.